modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/leases/LeaseTracker.java

```diff
@@ -79,47 +89,33 @@ public class LeaseTracker implements PlacementDriver {
     /** Listener to update a leases cache. */
     private final UpdateListener updateListener = new UpdateListener();
 
+    /** Versioned value used only at manager startup to correctly fire table creation events. */
+    private final IncrementalVersionedValue<Void> startVv;
+
     /**
      * Constructor.
      *
+     * @param registry Registry for versioned values.
      * @param msManager Meta storage manager.
      */
-    public LeaseTracker(MetaStorageManager msManager) {
+    public LeaseTracker(Consumer<LongFunction<CompletableFuture<?>>> registry, MetaStorageManager msManager) {
         this.msManager = msManager;
+
+        startVv = new IncrementalVersionedValue<>(registry);
     }
 
     /** Recoveries state from Vault and subscribers on further updates. */
     public void startTrack() {
```

Comment:
I would propose to make `startTrack` to return a `CompletableFuture` and then move the `startVv` to the upper component (`PlacementDriverManager`). This way we will only have IgniteComponents to worry about `startVv` which will help with later transition to async start

Reply:
Did it